src/app/static/src/js/store/git/actions.ts

```diff
@@ -5,17 +5,31 @@ import {GitMutation} from "./mutations";
 import {RunnerRootState} from "../../utils/types";
 
 export enum GitAction {
-    FetchMetadata = "FetchMetadata"
+    FetchMetadata = "FetchMetadata",
+    SelectBranch = "SelectBranch",
+    SelectCommit = "SelectCommit"
 }
 
-type GitActionHandler = (store: ActionContext<GitState, RunnerRootState>) => void
+type GitActionHandler<T> = (store: ActionContext<GitState, RunnerRootState>, payload: T) => void
 
-export const actions: ActionTree<GitState, RunnerRootState> & Record<GitAction, GitActionHandler> = {
+export const actions: ActionTree<GitState, RunnerRootState> & Record<GitAction, GitActionHandler<any>> = {
 
     async [GitAction.FetchMetadata](context) {
         await api.get('/report/run-metadata')
             .then(({data}) => {
                 context.commit(GitMutation.SetMetadata, data.data)
             })
+    },
+
+    async [GitAction.SelectBranch](context, selectedBranch: string) {
+        context.commit(GitMutation.SelectBranch, selectedBranch)
+        await api.get(`/git/branch/${selectedBranch}/commits/`)
+            .then(({ data }) => {
+                context.commit(GitMutation.SetCommits, data.data);
+            })
+    },
+
+    async [GitAction.SelectCommit](context, selectedCommit: string) {
```

Comment:
This doesn't need to access the api, so component could just commit the mutation directly without needing this action.